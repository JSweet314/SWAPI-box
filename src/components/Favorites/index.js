import React from 'react';
import PropTypes from 'prop-types';
import PersonCard from '../PersonCard/index';
import VehicleCard from '../VehicleCard/index';
import PlanetCard from '../PlanetCard/index';
import './style.css';

const Favorites = ({favorites, handleFavoriteClick}) => {
  const favoriteCards = favorites.map(favorite => {
    switch (favorite.category) {
    case 'people': 
      return <PersonCard
        handleFavoriteClick={handleFavoriteClick}
        favorites={favorites}
        key={favorite.name}
        card={favorite} />;
    case 'vehicles':
      return <VehicleCard
        handleFavoriteClick={handleFavoriteClick}
        favorites={favorites}
        key={favorite.name}
        card={favorite} />;
    case 'planets':
      return <PlanetCard
        handleFavoriteClick={handleFavoriteClick}
        favorites={favorites}
        key={favorite.name}
        card={favorite} />;
    default:
      return null;
    }
  });

  if (!favorites.length) {
    return (
      <div className="favorites-display">
        <h5>You Have No Favorites</h5>
      </div>
    );
  }

  return (
    <div className="favorites-display">
      {favoriteCards}
    </div>
  );
};

Favorites.propTypes = {
  favorites: PropTypes.array.isRequired,
  handleFavoriteClick: PropTypes.func.isRequired
};

export default Favorites;