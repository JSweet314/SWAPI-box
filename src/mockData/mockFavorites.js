export const mockFavorites = [
  {
    category: "planets",
    climate: "temperate, tropical",
    name: "Yavin IV",
    population: "1000",
    residents: [],
    terrain: "jungle, rainforests"
  },
  {
    category: "vehicles",
    model: "Digger Crawler",
    name: "Sand Crawler",
    numberOfPassengers: "30",
    vehicleClass: "wheeled"
  },
  {
    category: "people",
    homeworld: "Tatooine",
    name: "Luke Skywalker",
    population: "200000",
    species: "Human"
  }
];