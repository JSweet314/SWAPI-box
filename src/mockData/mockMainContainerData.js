/* eslint-disable no-undef */
export const mockGetDataByRouteId = jest.fn();
export const mockGetPeopleData = jest.fn();
export const mockGetPlanetsData = jest.fn();
export const mockGetVehiclesData = jest.fn();
export const mockGetPreviousPage = jest.fn();
export const mockGetNextPage = jest.fn();
export const mockPersonCards = jest.fn();
export const mockPlanetCards = jest.fn();
export const mockVehicleCards = jest.fn();
export const mockBuildCards = jest.fn();
export const mockDeployPeopleData = jest.fn();
export const mockDeployPlanetsData = jest.fn();
export const mockDeployVehiclesData = jest.fn();
export const mockInitialMainState = {
  people: [],
  planets: [],
  vehicles: [],
  next: null,
  previous: null,
  loading: false,
  errorStatus: ''
};
export const mockMainProps = {
  favorites: [],
  handleFavoriteClick: jest.fn(),
  match: {
    isExact: false,
    params: { id: 'people' },
    path: "/category/:id",
    url: "/category/people"
  }
};
export const mockMainPropsPlanetsRoute = {
  favorites: [],
  handleFavoriteClick: jest.fn(),
  match: {
    isExact: false,
    params: { id: 'planets' },
    path: "/category/:id",
    url: "/category/people"
  }
};
export const mockMainPropsVehiclesRoute = {
  favorites: [],
  handleFavoriteClick: jest.fn(),
  match: {
    isExact: false,
    params: { id: 'vehicles' },
    path: "/category/:id",
    url: "/category/people"
  }
};
export const mockPersonCard = {
  homeworld: "Tatooine",
  name: "Luke Skywalker",
  population: "200000",
  species: "Human"
};
export const mockPlanetCard = {
  name: "Alderaan",
  population: "2000000000",
  terrain: "grasslands, mountains",
  climate: "temperate",
  residents: ["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"]
};
export const mockVehicleCard = {
  name: "Sand Crawler",
  model: "Digger Crawler",
  vehicleClass: "wheeled",
  numberOfPassengers: "30"
};
export const mockStoragePeople = {
  next: "https://swapi.co/api/people/?format=json&page=2",
  previous: null,
  dataArray: [mockPersonCard]
};
export const mockStoragePlanets = {
  next: "https://swapi.co/api/planets/?format=json&page=2",
  previous: null,
  dataArray: [mockPlanetCard]
};
export const mockStorageVehicles = {
  next: "https://swapi.co/api/vehicles/?format=json&page=2",
  previous: null,
  dataArray: [mockVehicleCard]
};
export const mockPeopleData = {
  next: "https://swapi.co/api/people/?format=json&page=2",
  previous: null,
  peopleArray: [mockPersonCard]
};
export const mockPlanetsData = {
  next: "https://swapi.co/api/planets/?format=json&page=2",
  previous: null,
  planetsArray: [mockPlanetCard]
};
export const mockVehiclesData = {
  next: "https://swapi.co/api/vehicles/?format=json&page=2",
  previous: null,
  vehiclesArray: [mockVehicleCard]
};