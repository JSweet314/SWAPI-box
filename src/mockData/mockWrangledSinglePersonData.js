export const mockWrangledSinglePersonData = {
  name: "Luke Skywalker",
  homeworld: "Tatooine",
  population: "200000",
  species: "https://swapi.co/api/species/?format=json&page=1"
};
