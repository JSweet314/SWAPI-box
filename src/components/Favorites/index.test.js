import React from 'react';
import {shallow} from 'enzyme';
import Favorites from './index';
import {mockFavorites} from '../../mockData/mockFavorites';

describe('Favorites', () => {
  /*eslint-disable no-undef*/
  const mockHandleFavoriteClick = jest.fn();
  /*eslint-disable no-undef*/

  it('should match a snapshot without favorites', () => {
    const wrapper = shallow(
      <Favorites
        favorites={[]}
        handleFavoriteClick={mockHandleFavoriteClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot with favorites', () => {
    const wrapper = shallow(
      <Favorites
        favorites={mockFavorites}
        handleFavoriteClick={mockHandleFavoriteClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});