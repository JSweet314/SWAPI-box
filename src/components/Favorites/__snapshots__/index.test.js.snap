// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match a snapshot with favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={
        Array [
            Object {
              "category": "planets",
              "climate": "temperate, tropical",
              "name": "Yavin IV",
              "population": "1000",
              "residents": Array [],
              "terrain": "jungle, rainforests",
            },
            Object {
              "category": "vehicles",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "numberOfPassengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "category": "people",
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ]
    }
    handleFavoriteClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PlanetCard
          card={
                    Object {
                              "category": "planets",
                              "climate": "temperate, tropical",
                              "name": "Yavin IV",
                              "population": "1000",
                              "residents": Array [],
                              "terrain": "jungle, rainforests",
                            }
          }
          favorites={
                    Array [
                              Object {
                                "category": "planets",
                                "climate": "temperate, tropical",
                                "name": "Yavin IV",
                                "population": "1000",
                                "residents": Array [],
                                "terrain": "jungle, rainforests",
                              },
                              Object {
                                "category": "vehicles",
                                "model": "Digger Crawler",
                                "name": "Sand Crawler",
                                "numberOfPassengers": "30",
                                "vehicleClass": "wheeled",
                              },
                              Object {
                                "category": "people",
                                "homeworld": "Tatooine",
                                "name": "Luke Skywalker",
                                "population": "200000",
                                "species": "Human",
                              },
                            ]
          }
          handleFavoriteClick={[Function]}
/>,
        <VehicleCard
          card={
                    Object {
                              "category": "vehicles",
                              "model": "Digger Crawler",
                              "name": "Sand Crawler",
                              "numberOfPassengers": "30",
                              "vehicleClass": "wheeled",
                            }
          }
          favorites={
                    Array [
                              Object {
                                "category": "planets",
                                "climate": "temperate, tropical",
                                "name": "Yavin IV",
                                "population": "1000",
                                "residents": Array [],
                                "terrain": "jungle, rainforests",
                              },
                              Object {
                                "category": "vehicles",
                                "model": "Digger Crawler",
                                "name": "Sand Crawler",
                                "numberOfPassengers": "30",
                                "vehicleClass": "wheeled",
                              },
                              Object {
                                "category": "people",
                                "homeworld": "Tatooine",
                                "name": "Luke Skywalker",
                                "population": "200000",
                                "species": "Human",
                              },
                            ]
          }
          handleFavoriteClick={[Function]}
/>,
        <PersonCard
          card={
                    Object {
                              "category": "people",
                              "homeworld": "Tatooine",
                              "name": "Luke Skywalker",
                              "population": "200000",
                              "species": "Human",
                            }
          }
          favorites={
                    Array [
                              Object {
                                "category": "planets",
                                "climate": "temperate, tropical",
                                "name": "Yavin IV",
                                "population": "1000",
                                "residents": Array [],
                                "terrain": "jungle, rainforests",
                              },
                              Object {
                                "category": "vehicles",
                                "model": "Digger Crawler",
                                "name": "Sand Crawler",
                                "numberOfPassengers": "30",
                                "vehicleClass": "wheeled",
                              },
                              Object {
                                "category": "people",
                                "homeworld": "Tatooine",
                                "name": "Luke Skywalker",
                                "population": "200000",
                                "species": "Human",
                              },
                            ]
          }
          handleFavoriteClick={[Function]}
/>,
      ],
      "className": "favorites-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Yavin IV",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "category": "planets",
            "climate": "temperate, tropical",
            "name": "Yavin IV",
            "population": "1000",
            "residents": Array [],
            "terrain": "jungle, rainforests",
          },
          "favorites": Array [
            Object {
              "category": "planets",
              "climate": "temperate, tropical",
              "name": "Yavin IV",
              "population": "1000",
              "residents": Array [],
              "terrain": "jungle, rainforests",
            },
            Object {
              "category": "vehicles",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "numberOfPassengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "category": "people",
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ],
          "handleFavoriteClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Sand Crawler",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "category": "vehicles",
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "numberOfPassengers": "30",
            "vehicleClass": "wheeled",
          },
          "favorites": Array [
            Object {
              "category": "planets",
              "climate": "temperate, tropical",
              "name": "Yavin IV",
              "population": "1000",
              "residents": Array [],
              "terrain": "jungle, rainforests",
            },
            Object {
              "category": "vehicles",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "numberOfPassengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "category": "people",
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ],
          "handleFavoriteClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Luke Skywalker",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "category": "people",
            "homeworld": "Tatooine",
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          },
          "favorites": Array [
            Object {
              "category": "planets",
              "climate": "temperate, tropical",
              "name": "Yavin IV",
              "population": "1000",
              "residents": Array [],
              "terrain": "jungle, rainforests",
            },
            Object {
              "category": "vehicles",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "numberOfPassengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "category": "people",
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ],
          "handleFavoriteClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PlanetCard
            card={
                        Object {
                                    "category": "planets",
                                    "climate": "temperate, tropical",
                                    "name": "Yavin IV",
                                    "population": "1000",
                                    "residents": Array [],
                                    "terrain": "jungle, rainforests",
                                  }
            }
            favorites={
                        Array [
                                    Object {
                                      "category": "planets",
                                      "climate": "temperate, tropical",
                                      "name": "Yavin IV",
                                      "population": "1000",
                                      "residents": Array [],
                                      "terrain": "jungle, rainforests",
                                    },
                                    Object {
                                      "category": "vehicles",
                                      "model": "Digger Crawler",
                                      "name": "Sand Crawler",
                                      "numberOfPassengers": "30",
                                      "vehicleClass": "wheeled",
                                    },
                                    Object {
                                      "category": "people",
                                      "homeworld": "Tatooine",
                                      "name": "Luke Skywalker",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                  ]
            }
            handleFavoriteClick={[Function]}
/>,
          <VehicleCard
            card={
                        Object {
                                    "category": "vehicles",
                                    "model": "Digger Crawler",
                                    "name": "Sand Crawler",
                                    "numberOfPassengers": "30",
                                    "vehicleClass": "wheeled",
                                  }
            }
            favorites={
                        Array [
                                    Object {
                                      "category": "planets",
                                      "climate": "temperate, tropical",
                                      "name": "Yavin IV",
                                      "population": "1000",
                                      "residents": Array [],
                                      "terrain": "jungle, rainforests",
                                    },
                                    Object {
                                      "category": "vehicles",
                                      "model": "Digger Crawler",
                                      "name": "Sand Crawler",
                                      "numberOfPassengers": "30",
                                      "vehicleClass": "wheeled",
                                    },
                                    Object {
                                      "category": "people",
                                      "homeworld": "Tatooine",
                                      "name": "Luke Skywalker",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                  ]
            }
            handleFavoriteClick={[Function]}
/>,
          <PersonCard
            card={
                        Object {
                                    "category": "people",
                                    "homeworld": "Tatooine",
                                    "name": "Luke Skywalker",
                                    "population": "200000",
                                    "species": "Human",
                                  }
            }
            favorites={
                        Array [
                                    Object {
                                      "category": "planets",
                                      "climate": "temperate, tropical",
                                      "name": "Yavin IV",
                                      "population": "1000",
                                      "residents": Array [],
                                      "terrain": "jungle, rainforests",
                                    },
                                    Object {
                                      "category": "vehicles",
                                      "model": "Digger Crawler",
                                      "name": "Sand Crawler",
                                      "numberOfPassengers": "30",
                                      "vehicleClass": "wheeled",
                                    },
                                    Object {
                                      "category": "people",
                                      "homeworld": "Tatooine",
                                      "name": "Luke Skywalker",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                  ]
            }
            handleFavoriteClick={[Function]}
/>,
        ],
        "className": "favorites-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Yavin IV",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "category": "planets",
              "climate": "temperate, tropical",
              "name": "Yavin IV",
              "population": "1000",
              "residents": Array [],
              "terrain": "jungle, rainforests",
            },
            "favorites": Array [
              Object {
                "category": "planets",
                "climate": "temperate, tropical",
                "name": "Yavin IV",
                "population": "1000",
                "residents": Array [],
                "terrain": "jungle, rainforests",
              },
              Object {
                "category": "vehicles",
                "model": "Digger Crawler",
                "name": "Sand Crawler",
                "numberOfPassengers": "30",
                "vehicleClass": "wheeled",
              },
              Object {
                "category": "people",
                "homeworld": "Tatooine",
                "name": "Luke Skywalker",
                "population": "200000",
                "species": "Human",
              },
            ],
            "handleFavoriteClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Sand Crawler",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "category": "vehicles",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "numberOfPassengers": "30",
              "vehicleClass": "wheeled",
            },
            "favorites": Array [
              Object {
                "category": "planets",
                "climate": "temperate, tropical",
                "name": "Yavin IV",
                "population": "1000",
                "residents": Array [],
                "terrain": "jungle, rainforests",
              },
              Object {
                "category": "vehicles",
                "model": "Digger Crawler",
                "name": "Sand Crawler",
                "numberOfPassengers": "30",
                "vehicleClass": "wheeled",
              },
              Object {
                "category": "people",
                "homeworld": "Tatooine",
                "name": "Luke Skywalker",
                "population": "200000",
                "species": "Human",
              },
            ],
            "handleFavoriteClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Luke Skywalker",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "category": "people",
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
            "favorites": Array [
              Object {
                "category": "planets",
                "climate": "temperate, tropical",
                "name": "Yavin IV",
                "population": "1000",
                "residents": Array [],
                "terrain": "jungle, rainforests",
              },
              Object {
                "category": "vehicles",
                "model": "Digger Crawler",
                "name": "Sand Crawler",
                "numberOfPassengers": "30",
                "vehicleClass": "wheeled",
              },
              Object {
                "category": "people",
                "homeworld": "Tatooine",
                "name": "Luke Skywalker",
                "population": "200000",
                "species": "Human",
              },
            ],
            "handleFavoriteClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Favorites should match a snapshot without favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={Array []}
    handleFavoriteClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h5>
        You Have No Favorites
</h5>,
      "className": "favorites-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "You Have No Favorites",
      },
      "ref": null,
      "rendered": "You Have No Favorites",
      "type": "h5",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h5>
          You Have No Favorites
</h5>,
        "className": "favorites-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You Have No Favorites",
        },
        "ref": null,
        "rendered": "You Have No Favorites",
        "type": "h5",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
